### PR TITLE
Only show party list once post is known

### LINKED ISSUE #795: GB party list shown initially when adding NI election
When adding a Northern Ireland election to a candidate, the Great Britain party dropdown is displayed initially:

image

After selecting a post, the correct party list is displayed:

image

### PR DESCRIPTION
**Pull request by symroe**
This is because the party register can change depending on the post.

Previous to this change we used to show GB parties by default, but that
was confusing when adding candidates in a post with the NI register

Fixes #795

### CODE REVIEW

**Comment by chris48s** [comment]
I am confused about this for two reasons.

1. This is now showing me 2 party drop downs before I've selected a post:

GIFrecord_2019-03-18_182026


2. Why do we need to wait until the post is selected anyway? We can work it out from the election object. For example, if I hit `bulk_adding/party/local.derry-city-and-strabane.2019-05-02/`, it knows which party list to show me without having to select a post first.

**Comment by symroe** [comment]
Hmm, this looks like something's up with point 1. I can't reproduce it – I'll look in to why that might be in the morning.

> We can work it out from the election object

Not so! Or at least not all the time. Consider `parl.2017-06-08`. The _election_ object contains _posts_ across the whole of the UK, but some of them use a different party register. I expect EU would be the same, however I think that's the only two instances where an election spans registers.

The correct model is to use the post or ballot to define the register, not the election.

**Comment by chris48s** [comment]
Aah yes. I was trying to think of a counter-example but couldn't.. I always get in a mess trying to work out which group level becomes an `Election` in YNR :man_facepalming:  . Interestingly, if I hit `bulk_adding/party/parl.2017-06-08/` that gives me 2 party lists, so I assume its looking at all the PEEs in the election group to work out if they are all GB, all NI or split.. Either way, yeah - waiting for the post/PEE is probably sensible given there are election groups which can cover GB/NI.

To reproduce the first issue, I just hit http://127.0.0.1:8000/election/local.windsor-and-maidenhead.2019-05-02/person/create/ and it happens. This was the only case I checked as I assume this is the only instance of this form where its really relevant because AFAIK in all the other places we add a person we select the post/PEE before we hit the person form anyway.

**Comment by symroe** [comment]
> To reproduce the first issue

So I only see a single party drop down. That might be wrong too, but let's focus for now on why you're seeing two. Do you mind re-importing the parties(`parties_import_from_ec`) and elections (`uk_create_elections_from_every_election`) on your local copy?

**Comment by chris48s** [comment]
I tried re-running the parties and elections imports. I've checked I've definitely got the same js code as you in `person_form.js` and I've checked there are no JS errors on the console. Still getting the same behaviour.

Suspect it probably doesn't help, but the 2 `<select>`s I've got are

`<select name="party_NI_local.windsor-and-maidenhead.2019-05-02" class="party-select party-select-local.windsor-and-maidenhead.2019-05-02" id="id_party_NI_local.windsor-and-maidenhead.2019-05-02">`

`<select name="party_GB_local.windsor-and-maidenhead.2019-05-02" class="party-select party-select-local.windsor-and-maidenhead.2019-05-02" id="id_party_GB_local.windsor-and-maidenhead.2019-05-02">`

Also, I get the same behaviour for a NI election as I do for a GB one.

**Comment by symroe** [comment]
[GB party list shown initially when adding NI election https://github.com/DemocracyClub/yournextrepresentative/issues/795](https://trello.com/c/ZDIY9g4t/1702-gb-party-list-shown-initially-when-adding-ni-election-https-githubcom-democracyclub-yournextrepresentative-issues-795)

**Comment by symroe** [comment]
Ok I had a think about this. 

Firstly, I've pushed a change that deals with the case you raised. That is, only show the party drop downs for the registers the election covers. For `local.windsor-and-maidenhead.2019-05-02` this is only GB, so in that sense the bug is "fixed". 

However, it would still be a problem for any elections that span registers (General elections, EU, maybe others).

I'm starting to think that the solution to this is to move to only _ever_ showing one drop down and populating them with all parties for that election. We can add the register to the party names and validate the selection on submit when we have the ballot.

In JS we can filter out / disable any parties that don't match the register for the selected ballot.

I think this is a larger change that I'd like to punt to a different issue, especially given that we don't (yet) have any elections that would cause this problem to materialise.

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging f42a897c5a3a6ac2dfe6e59f5c859ad7be683c87 into baa56423f7bf44e9e2a0491327a4d747692590b8 - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-aaca8b4f38acb66f1d8fdd084e1241028fda4684)

**fixed alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging c855fd28464417372dec827147bd46961d6fe47d into db4a6df3376258739beb38806b4ad223f32bce98 - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-4556dddf53fc58ef494fd72d0fe2ca4f9c30560f)

**fixed alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by chris48s** [comment]
This seems to break bulk add by party e.g:

http://127.0.0.1:8000/bulk_adding/party/local.derry-city-and-strabane.2019-05-02/

**Comment by symroe** [comment]
Can you explain what's broken?

I see this:

Screenshot 2019-03-29 20 15 03

**Comment by chris48s** [comment]
Sorry - that was a really unhelpful review comment. For me, bulk add by party is working as expected for a GB election, but I'm getting no party list on bulk add by party for a NI election:

Screenshot at 2019-03-31 16-34-33

Screenshot at 2019-03-31 16-34-47

If I reload the page, I can see that some drop-downs are initially loaded and then hidden.

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging 073a76edd008659bcba93357de8fad1d59b74f73 into e3a31306423cac2c71535a52f8ded66a924055ba - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-d454c7fd1351a0f290645efdc17bfd453178bd6e)

**fixed alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by symroe** [comment]
After chatting on Slack, Chris pointed out this (obvious when looking) bug.

This should solve the "`local.derry-city-and-strabane.2019-05-02`" problem.

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging 26284d2104f00e8e88bc9113ab2cccb20b98a9ea into e3a31306423cac2c71535a52f8ded66a924055ba - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-8044022645b6cfa1444883ec6e75035b3bd9b93e)

**fixed alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: ynr/apps/bulk_adding/templates/bulk_add/sopns/add_form.html [hunk 1]
```diff
@@ -39,6 +39,23 @@ <h2>Add candidates to {{ election_obj.name }}: <a href="{{ post_election.get_abs
   </ol>
   </details>
 
+  {% if post_election.rawpeople and post_election.rawpeople.source_type == "parsed_pdf"%}
+    <div class="panel callout">
+      <details open>
+        <summary>
+          🤖 This SoPN has been parsed by a bot 🤖
```
Comment (chris48s) [inline ynr/apps/bulk_adding/templates/bulk_add/sopns/add_form.html]: I think this file belongs in another PR :)

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging 8079e1e2391edcd0b18996a6b6ed1f81ff181dcb into ac39275efab5ececd0e2cfa06972d3623ff11750 - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-ac9b585f2bf686ebe003f491f76c6d84500c01fb)

**fixed alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*
